internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -47,6 +48,206 @@ func New(sess *session.Session) CloudFormation {
+	// new accounts passed in as environments.
+	accounts := map[string]bool{}
+	for _, e := range envs {
+		accounts[e.AccountID] = true
+	}
+
+	for _, previouslyDeployedAcct := range previouslyDeployedConfig.Accounts {
+		accounts[previouslyDeployedAcct] = true
+	}
+
+	accountList := []string{}
+	for accountID := range accounts {
+		accountList = append(accountList, accountID)
+	}
+
+	// Create a set of apps we need to include in the project config
+	// including those which were previously deployed and the potential
+	// new apps passed in.
+	appsToDeploy := map[string]bool{}
+	for _, app := range apps {
+		appsToDeploy[app.Name] = true
+	}
+
+	for _, previouslyDeployedApp := range previouslyDeployedConfig.Apps {
+		appsToDeploy[previouslyDeployedApp] = true
+	}
+
+	appList := []string{}
+	for app := range appsToDeploy {
+		appList = append(appList, app)
+	}
+
+	// Every time we deploy the StackSet, we include a version field in the stack metadata.
+	// When we go to update the StackSet, we include that version + 1 as the "Operation ID".
+	// This ensures that we don't overwrite any changes that may have been applied between
+	// us reading the stack and actually updating it.
+	// As an example:
+	//  * We read the stack with Version 1
+	//  * Someone else reads the stack with Version 1
+	//  * We update the StackSet with Version 2, the update completes.
+	//  * Someone else tries to update the StackSet with their stale version 2.
+	//  * "2" has already been used as an operation ID, and the stale write fails.
+	newDeploymentConfig := projectResourcesConfig{
+		Version:  previouslyDeployedConfig.Version + 1,
+		Apps:     appList,
+		Accounts: accountList,
+		Project:  projectConfig.Project.Name,
+	}
+
+	newTemplateToDeploy, err := projectConfig.ResourceTemplate(&newDeploymentConfig)
+	if err != nil {
+		return err
+	}
+	input := cloudformation.UpdateStackSetInput{
+		TemplateBody:          aws.String(newTemplateToDeploy),
+		OperationId:           aws.String(fmt.Sprintf("%d", newDeploymentConfig.Version)),
+		StackSetName:          aws.String(projectConfig.StackSetName()),
+		Description:           aws.String(projectConfig.StackSetDescription()),
+		ExecutionRoleName:     aws.String(projectConfig.StackSetExecutionRoleName()),
+		AdministrationRoleARN: aws.String(projectConfig.StackSetAdminRoleARN()),
+		Tags:                  projectConfig.Tags(),
+	}
+	output, err := cf.client.UpdateStackSet(&input)
+
+	if err != nil {
+		return fmt.Errorf("updating project resource template %s", err)
+	}
+
+	return cf.waitForStackSetOperation(projectConfig.StackSetName(), *output.OperationId)
+}
+
+// addNewStackInstances takes a list of environments and calculates if it needs to create a new
+// stack instance. This is done by determining if an environment is in a new region.
+func (cf CloudFormation) addNewStackInstances(projectConfig *projectStackConfig, envs []*archer.Environment) error {
```

Comment:
same over here regarding the method name, `updateProjectWithNewRegions` maybe?

Reply:
I reorganized it a bit, let me know what you think!